### PR TITLE
App ws auth

### PR DESCRIPTION
**Pull request by ThetaSinner**

### CODE REVIEW

File: test/e2e/index.ts [hunk 1]
```diff
@@ -1298,28 +1287,3 @@ test(
     t.equal(response, "hi", "updated coordinator zomes can be called");
   })
 );
-
-test(
```
Comment (ThetaSinner) [inline test/e2e/index.ts:1302]: Couldn't find a sensible way to retain this. I think if the websocket connection drops then the app needs to reconnect

File: .github/workflows/integration-test.yml [hunk 2]
```diff
@@ -65,7 +65,7 @@ jobs:
         run: nix develop -c $SHELL -c "npm ci"
 
       - name: Run tests
-        run: nix develop -c $SHELL -c "npm t"
+        run: nix develop --override-input versions/holochain "github:holochain/holochain/holochain-0.3.0-beta-dev.48" -c $SHELL -c "npm t"
```
Comment (ThetaSinner) [inline .github/workflows/integration-test.yml:68]: Until Holonix is updated we need to test against a specific Holochain that is newer than the one in weekly.

As we work up to automatically testing new releases before releasing Holonix updates, we should find a way to make this simpler than changing the workflow file... Not sure how!

File: package.json [hunk 3]
```diff
@@ -32,7 +32,8 @@
   ],
   "scripts": {
     "lint": "eslint --fix --ext .ts src test .eslintrc.cjs",
-    "test:app-agent": "RUST_LOG=error RUST_BACKTRACE=1 tsx test/e2e/app-agent-websocket.ts",
+    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
```
Comment (jost-s) [inline package.json:35]: 💯

File: src/api/client.ts [hunk 4]
```diff
@@ -159,38 +165,48 @@ export class WsClient extends Emittery {
...
+   * @param request The authentication request, containing an app authentication token.
+   */
+  async authenticate(request: AppAuthenticationRequest): Promise<void> {
+    return this.exchange(request, (request, resolve) => {
+      const encodedMsg = encode({
+        type: "authenticate",
+        data: encode(request),
+      });
+      this.socket.send(encodedMsg);
+      // Message just needs to be send first, no need to wait for a response or even require a flush
```
Comment (jost-s) [inline src/api/client.ts]:
```suggestion
      // Message just needs to be sent first, no need to wait for a response or even require a flush
```

File: src/api/common.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ export type Requester<Req, Res> = (req: Req, timeout?: number) => Promise<Res>;
 /**
  * @public
  */
-export type RequesterUnit<Res> = () => Promise<Res>;
+export type RequesterNoArg<Res> = (timeout?: number) => Promise<Res>;
```
Comment (jost-s) [inline src/api/common.ts:22]: Is this a rename from Unit (Rust) to NoArg (JS)? Now it has an arg ;-) Anyhow, not that important

File: src/api/app/types.ts [hunk 6]
```diff
@@ -194,12 +262,26 @@ export type NetworkInfoResponse = NetworkInfo[];
...
  */
-export interface AppApi {
-  appInfo: Requester<AppInfoRequest, AppInfoResponse>;
-  callZome: Requester<CallZomeRequest, CallZomeResponse>;
-  enableCloneCell: Requester<EnableCloneCellRequest, EnableCloneCellResponse>;
-  disableCloneCell: Requester<
-    DisableCloneCellRequest,
-    DisableCloneCellResponse
-  >;
+export interface AppAgentClient {
```
Comment (jost-s) [inline src/api/app/types.ts]: I think we can rename all `AppAgent*` to `App*`. WDYT?

File: src/api/app/types.ts [hunk 7]
```diff
@@ -1,22 +1,102 @@
...
+ */
+export type RoleNameCallZomeRequestSigned = Omit<
+  CallZomeRequestSigned,
+  "cell_id"
+> & { role_name: RoleName };
+
+/**
+ * @public
+ */
+export type AppAgentCallZomeRequest =
```
Comment (jost-s) [inline src/api/app/types.ts]: rename here too?

**Review by jost-s — APPROVED** [review]
Groovy!

File: src/api/common.ts (hunk 5 above)
Comment (ThetaSinner) [reply src/api/common.ts:22]: Thanks diff tool! `RequesterUnit` wasn't used so I deleted it. This one is supposed to be "no request argument"

File: src/api/app/types.ts (hunk 6 above)
Comment (ThetaSinner) [reply src/api/app/types.ts]: Heck, had moved this stuff without looking too closely - hadn't realised how many instances there were of 'agent' still. On it!

**Pull request merged**